src/rpc/mining.cpp

```diff
@@ -734,6 +732,7 @@ static RPCHelpMan getblocktemplate()
             if (!DecodeHexBlk(block, dataval.get_str()))
                 throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Block decode failed");
 
+            LOCK(cs_main);
```

Comment:
nit: could be one line lower to reduce scope more

Reply:
Makes sense to me! updated in [493ba0f](https://github.com/bitcoin/bitcoin/pull/33264/commits/493ba0f688311280056491986068cd4a9ad2be66)